config/routes.rb

```diff
@@ -26,7 +26,9 @@
     resources :users, only: [:index, :show]
 
     resources :posts do
-      resources :comments, module: :posts
+      resources :comments, module: :posts do
+        get 'page/:page', action: :index, on: :collection
```

Comment:
do you really need this route? I think you can do it through parameters in URL - `comments?page=1`